src/extension.ts

```diff
@@ -1941,12 +1941,30 @@ function isLintable(document: vscode.TextDocument): boolean {
     return document.languageId === 'yaml' || document.languageId === 'json' || document.languageId === 'helm';
 }
 
+function linterEnabled(enabledLinters: string[], name: string): boolean {
+    for (const linter of enabledLinters) {
+        if (linter === name) {
+            return true;
+        }
+    }
+    return false;
+}
+
 async function kubernetesLint(document: vscode.TextDocument): Promise<void> {
     // Is it a Kubernetes document?
     if (!isLintable(document)) {
         return;
     }
-    const linterPromises = linters.map((l) => l.lint(document));
+    const enabledLinters = config.getEnabledLinters();
+    if (enabledLinters.length === 0) {
+        return;
+    }
+    const linterPromises: Promise<vscode.Diagnostic[]>[] = [];
+    linters.map((l) => {
+        if (linterEnabled(enabledLinters, l.name())) {
+            linterPromises.push(l.lint(document));
+        }
+    });
```

Comment:
Depending on your config defaults it looks like this requires users to opt into every linter.  This makes new linters essentially useless as nobody will ever read the docs to find out about them.  It may also vex Seans Knox and McKenna, who asked for the resource limits linter because people were shooting themselves in the foot deploying unrestricted pods to AKS and then raising starvation as a service issue.

A preferable user interface would be:

* Linters are **on by default**.
* You can deal with a lint warning by fixing it, by disabling the linter at workspace level, or by disabling the linter at global level.

I would prefer not to have the option to disable all (present and future) linting, but if others disagree then we can have that as a manual setting (_not_ exposed as a diagnostic fix in the UI).

Reply:
Fixed this so the list is ones to disable, not ones to enable, and the default is all on.

I don think we want a "make it stop" option. 

With things like linting someone is always going to be annoyed, we need them to stop the bleeding as much as they can.